Default values for input object types are now displayed in Doc Explorer

excitement-engineer opened the pull request:
GraphiQL can now show default values for input object types. Currently, GraphiQL only shows default value for args that are scalar values, not input objects. This PR aims to solve this problem.

I think this solution is especially useful for a relay-compliant schemas where mutations are performed using a single `input`object argument.

Let me know what you think!

Here is a screenshot of what it looks like with the new addition:

<img width="352" alt="screen shot 2017-09-09 at 10 19 48" src="https://user-images.githubusercontent.com/19558321/30238516-f751afe0-9548-11e7-9e40-16730d2e1224.png">

asiandrummer commented on test/schema.js:
```diff
@@ -28,12 +28,12 @@ const TestEnum = new GraphQLEnumType({
     RED: { description: 'A rosy color' },
     GREEN: { description: 'The color of martians and slime' },
-    BLUE: { description: 'A feeling you might have if you can\'t use GraphQL' },
+    BLUE: { description: "A feeling you might have if you can't use GraphQL" },
```
nit: `'` instead

excitement-engineer replied:
The project has lint-staged enabled, this is changing the single quote to a double quote each time I commit. Given all the changes in the diff seems that prettier has not been run on all the files yet. I think that it's better to open another PR where prettier is run on the entire project so that new PRs will have clean diffs.

asiandrummer commented on src/components/DocExplorer/DefaultValue.js:
```diff
@@ -0,0 +1,31 @@
+      <span>
+        {' = '}
+        <span className="arg-default-value">
+          {print(astFromValue(defaultValue, type))}
```
Does `print` function always print with quotes around the default value? Could we check this with some more test cases, maybe for scalar types at least? (e.g. boolean, int, etc)

Also, the color difference slightly bothers me :D I'm even for using another color hmm... is `arg-default-value` an existing classname?

excitement-engineer replied:
I will add some test cases! This code is similar to the code uthat is currently used to display default values for scalar arguments. The classname that I ised was the same as before. I will take a look to see if there is a more suited color for the default value

excitement-engineer replied:
I ran prettier over the files in the test folders using PR #608.

asiandrummer commented:
@excitement-engineer Sorry we've been stalled giving you reviews! Lets work together on this next day or so to get this in! I'm super excited about it.

excitement-engineer commented:
Thanks for taking a look! I will update the pr as soon as I find some tome!

excitement-engineer commented:
Hey @asiandrummer I updated the PR. I added some test cases for boolean and integers and I can confirm that no quotes are printed for the default values. 

I also changed the color of the default value to green to increase the contrast between the field and the default value. See the screenshots below. 

Let me know what you think about the changes!

<img width="351" alt="screen shot 2017-09-30 at 15 04 21" src="https://user-images.githubusercontent.com/19558321/31046005-1a8549fa-a5f1-11e7-97f6-0a6b93ed14fd.png">
<img width="346" alt="screen shot 2017-09-30 at 15 04 39" src="https://user-images.githubusercontent.com/19558321/31046006-1b6417ac-a5f1-11e7-96c0-bbbb1df06a40.png">

asiandrummer commented:
cc @wincent @leebyron for visibility. This PR looks good to me, thanks again for your contribution @excitement-engineer!
